mantidproject/mantid
# MD Algorithm to convert from Q to HKL

**Description of work.**

Creates a new algorithm `ConvertQtoHKLMDHisto` which converts a `MDEventWorkspace` in QSample to a `MDHistoWorkspace` in HKL. The conversion can use the UB matrix from the input workspace, or from a `PeaksWorkspace`. 

**To test:**

Verify the unit test `ConvertQtoHKLMDHistoTest` passes.

The following script can be used to make sure that the conversion to HKL looks correct and that peaks are overlaid properly.
```
data = HB3AAdjustSampleNorm(Filename="/HFIR/HB3A/IPTS-19962/shared/autoreduce/HB3A_exp0737_scan0078.nxs")

peaks = HB3AFindPeaks(InputWorkspace=data, 
                      CellType="Tetragonal",
                      Centering="I",
                      PeakDistanceThreshold=0.25, 
                      DensityThresholdFactor=5000,
                      Wavelength=1.551)

TransformHKL(PeaksWorkspace=peaks, HKLTransform="-0.3333333,0,0,0,0.3333333,0,0,0,-1")

mdhist = ConvertQtoHKLMDHisto(InputWorkspace=data,
                              PeaksWorkspace=peaks)
```
When plotting in SliceViewer and overlaying the peaks, the peaks should appear correctly over the data:

![scd141_scan78_hkl_peaks](https://user-images.githubusercontent.com/6309092/101688989-090fcd00-3a2a-11eb-92f8-1c669918eb29.png)


*There is no associated issue.*



The version of this into `ornl-next` is #30136 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
